core/src/main/kotlin/plugins/Lifecycle.kt

```diff
@@ -42,8 +44,10 @@ fun Application.configureLifecycle() {
     environment.monitor.subscribe(DatabaseReady) {
         val authorizationService by inject<AuthorizationService>()
 
-        runBlocking(Dispatchers.IO) {
-            syncRoles(authorizationService)
+        thread {
```

Comment:
Obviously, it would make more sense to turn the calling function into a suspend function so just `runBlocking` is dropped, but I guess that's not easily possible for some reason? Could you provide a brief rationale in the commit message?